tests/Doctrine/Tests/ORM/Functional/OneToManyToOneOrphanRemovalTest.php

```diff
@@ -0,0 +1,269 @@
+
+        $reviews = new Reviews();
+        $reviews->add($review1);
+        $reviews->add($review2);
+        $reviews->add($review3);
+        $reviews->add($review4);
+
+        $replaceReviews                            = new Reviews();
+        $modifiedReviewFromOuterApplicationContext = new Review('review3', 'review 3 updated', 2, $user3);
+        $newReview5                                = new Review('review5', 'review 5', 2, $user3);
+        $replaceReviews->add($review1);
+        $replaceReviews->add($modifiedReviewFromOuterApplicationContext);
+        $replaceReviews->add($review4);
+        $replaceReviews->add($newReview5);
+        $this->replaceReviews = $replaceReviews;
+
+        $business = new Business('business1', 'business One', $reviews);
+
+        $this->_em->persist($business);
+        $this->_em->flush();
+
+        $this->businessId = $business->getId();
+
+        $this->_em->clear();
+    }
+
+    public function testOrphanRemoval(): void
+    {
+        $businessProxy = $this->_em->getReference(Business::class, $this->businessId);
+
+        $this->_em->remove($businessProxy);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $query  = $this->_em->createQuery('SELECT b FROM Doctrine\Tests\Models\Rating\Business b');
+        $result = $query->getResult();
+
+        self::assertCount(0, $result, 'Business should be removed by EntityManager');
+
+        $query  = $this->_em->createQuery('SELECT r FROM Doctrine\Tests\Models\Rating\Review r');
+        $result = $query->getResult();
+
+        self::assertCount(0, $result, 'Review should be removed by orphanRemoval');
+
+        $query  = $this->_em->createQuery('SELECT u FROM Doctrine\Tests\Models\Rating\User u');
+        $result = $query->getResult();
+
+        self::assertCount(0, $result, 'User should be removed by orphanRemoval');
+    }
+
+    public function testOrphanRemovalRemoveFromCollection(): void
+    {
+        $business = $this->_em->find(Business::class, $this->businessId);
+
+        $reviews = $business->getReviews();
+        $reviews->remove(1);
+        $reviews->removeElement($reviews->last());
+        $business->setReviews($reviews);
+
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $query  = $this->_em->createQuery('SELECT r FROM Doctrine\Tests\Models\Rating\Review r');
+        $result = $query->getResult();
+
+        self::assertCount(2, $result, 'Review should be removed by orphanRemoval');
+
+        $query  = $this->_em->createQuery('SELECT u FROM Doctrine\Tests\Models\Rating\User u');
+        $result = $query->getResult();
+
+        self::assertCount(2, $result, 'User should be removed by orphanRemoval');
+    }
+
+    public function testOrphanRemovalRemoveUpdateWithNewCollection(): void
+    {
+        $business = $this->_em->find(Business::class, $this->businessId);
+        $business->setReviews($this->replaceReviews);
```

Comment:
Cascade all does not detect existing user and does not do reconciliation
There was 1 error:

1) Doctrine\Tests\ORM\Functional\OneToManyToOneOrphanRemovalTest::testOrphanRemovalRemoveUpdateWithNewCollection
Doctrine\DBAL\Exception\UniqueConstraintViolationException: An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'user1' for key 'ratings_user.PRIMARY'